includes/class-wc-fgc-update-variation-cart.php

```diff
@@ -0,0 +1,387 @@
+		add_filter( 'wp_enqueue_scripts', array( __CLASS__, 'enqueue_cart_script' ) );
+
+		// Add edit option on the cart page.
+		add_action( 'woocommerce_after_cart_item_name', array( __CLASS__, 'variation_update_data' ), 1, 2 );
+
+		// Handle the ajax request of the update cart.
+		add_action( 'wp_ajax_wc_fgc_get_product_html', array( __CLASS__, 'get_variation_html' ) );
+		add_action( 'wp_ajax_nopriv_wc_fgc_get_product_html', array( __CLASS__, 'get_variation_html' ) );
+
+		// Update the cart as per the user choice.
+		add_action( 'wp_ajax_wc_fgc_update_variation_in_cart', array( __CLASS__, 'update_variation_in_cart' ) );
+		add_action( 'wp_ajax_nopriv_wc_fgc_update_variation_in_cart', array( __CLASS__, 'update_variation_in_cart' ) );
+
+		// Show error message on the checkout page.
+		add_action( 'woocommerce_before_checkout_process', array( __CLASS__, 'check_is_valid_product' ) );
+	}
+
+	/**
+	 * Enqueue needed js and css
+	 *
+	 * @name enqueue_cart_script
+	 * @param string $page   Name of the page.
+	 * @since 1.0.0
+	 */
+	public static function enqueue_cart_script( $page ) {
+
+		// check is cart page or not.
+		if ( is_cart() ) {
+			// Hook
+			add_filter( 'woocommerce_get_script_data', array( __CLASS__, 'change_variation_select_alert' ), 10, 2 );
+
+			$min_ext = ( ( defined( 'SCRIPT_DEBUG' ) && SCRIPT_DEBUG ) ? '' : 'min' );
+
+			// Enqueue default js for woocommerce.
+			wp_enqueue_script( 'wc-add-to-cart-variation' );
+
+			//Alter the add to cart variation text
+			// Enqueue js.
+			wp_enqueue_script( self::$name . '_js', plugin_dir_url( _WC_FGC_PLUGIN_NAME ) . 'assets/js/frontend-variation-cart' . $min_ext . '.js', array( 'jquery', 'flexslider', 'wc-single-product' ), self::$version, false );
+
+			$wc_fgc_trans_array = array(
+				'ajax_url'           => admin_url( 'admin-ajax.php' ), // ajax url.
+				'wc_fgc_nonce'       => wp_create_nonce( 'wc-fgc-verify-nonce' ),
+				'flexslider'         => apply_filters(
+					'woocommerce_single_product_carousel_options',
+					array(
+						'rtl'            => is_rtl(),
+						'animation'      => 'slide',
+						'smoothHeight'   => false,
+						'directionNav'   => false,
+						'controlNav'     => false, // set to false
+						'slideshow'      => false,
+						'animationSpeed' => 500,
+						'animationLoop'  => false, // Breaks photoswipe pagination if true.
+					)
+				),
+				'zoom_enabled'                => get_theme_support( 'wc-product-gallery-zoom' ),
+				'photoswipe_enabled'          => get_theme_support( 'wc-product-gallery-lightbox' ),
+				'flexslider_enabled'          => get_theme_support( 'wc-product-gallery-slider' ),
+				'server_error'                => __( '🤕 Sorry, an error occured, please try again later.', 'wc_free_gift_coupons'),
+			);
+
+			// Enqueue needed css for it.
+			wp_enqueue_style( self::$name . '_style', plugin_dir_url( _WC_FGC_PLUGIN_NAME ) . 'assets/css/frontend-variation-cart' . $min_ext . '.css', array(), self::$version, 'all' );
```

Comment:
can the name not be a variable?

Reply:
okay, was going to change that when merged with the main plugin, i think the dev did that for some reason.